packages/notionable/src/stores/root-store.ts

```diff
@@ -0,0 +1,82 @@
+
+import { NetworkClientModel } from '@notional-finance/core-entities';
+import { types, Instance, } from 'mobx-state-tree';
+import { PortfolioStoreModel } from './portfolio-store';
+import { getFromLocalStorage, Network, THEME_VARIANTS } from '@notional-finance/util';
+import { createContext } from 'react';
+import { AppStoreModel } from './app-store';
+
+
+export type RootStoreType = Instance<typeof RootStore>
+export type NetworkClientModelType = Instance<typeof NetworkClientModel>
+export type AppStoreModelType = Instance<typeof AppStoreModel>
+export type PortfolioStoreModelType = Instance<typeof PortfolioStoreModel>
+
+const userSettings = getFromLocalStorage('userSettings');
+
+const RootStore = types.model("RootStore", {
+  mainnetStore: NetworkClientModel, 
+  arbitrumStore: NetworkClientModel, 
+  allNetworksStore: NetworkClientModel, 
+  portfolioStore: PortfolioStoreModel,
+  appStore: AppStoreModel,
+}).actions(self => {
+  const getNetworkClient = (network: Network) => {
+    switch (network) {
+      case Network.mainnet:
+        return self.mainnetStore
+      case Network.arbitrum:
+        return self.arbitrumStore
+      default:
+        return self.allNetworksStore
+    }
+  }
+  return { getNetworkClient }
+})
+  export const createRootStore = (): RootStoreType => {
+    const rootStore = RootStore.create({
+      mainnetStore: {
+        network: Network.mainnet
+      },
+      arbitrumStore: {
+        network: Network.arbitrum
+      },
+      allNetworksStore: {
+        network: Network.all
+      },
+      portfolioStore: {
```

Comment:
When you define these stores, it's better to embed the default vault into the model definition using type.optional. Some of these values should also be undefined to signify that they haven't loaded, like `heroStats` for example.